pkg/detectors/github_oauth2/github_oauth2.go

```diff
@@ -20,6 +19,9 @@ type Scanner struct {
 
 // Ensure the Scanner satisfies the interfaces at compile time.
 var _ detectors.Detector = (*Scanner)(nil)
+var _ detectors.EndpointCustomizer = (*Scanner)(nil)
```

Comment:
Added this because a engine test case was failing due to missing method in this detector.

Reply:
hey @kashifkhan0771 would you mind double checking some comments [here](https://github.com/trufflesecurity/trufflehog/pull/3587#issuecomment-2471533293) around this. I might've misunderstood Miccah's comment and tried to fix this issue with: https://github.com/trufflesecurity/trufflehog/pull/3592

Not sure if I made it worse 😓

Reply:
Thank you for double checking. ❤️

Reply:
Done 🚀 Updated with your changes from main.